pytorch_lightning/trainer/trainer.py

```diff
@@ -2099,6 +2101,12 @@ def _exit_gracefully_on_signal(self) -> None:
             class_name = caller[0].f_locals["self"].__class__.__name__
             raise ExitGracefullyException(f"Exiting gracefully on {class_name}:{caller.function}")
 
+    @staticmethod
+    def _on_exit_gracefully_exception(exception: Exception) -> None:
+        if isinstance(exception, ExitGracefullyException):
+            # this is required to ensure schedulers can properly restart the training.
+            os._exit(0)
```

Comment:
minor suggestion:
- move this code into `_on_exception` method above
- in the test, mock.patch("pytorch_lightning.trainer.trainer.os._exit")  (it will be a no-op)
- optionally assert that `os._exit` was called

if it does not work, ignore my comment

Reply:
Donee !